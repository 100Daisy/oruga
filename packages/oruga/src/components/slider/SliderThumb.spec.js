import { shallowMount } from '@vue/test-utils'
import OSliderThumb from '@components/slider/SliderThumb'

let wrapper
const OSlider = {
    template: '<o-slider-stub></o-slider-stub>',
    data() {
        return {
            step: 5,
            precision: 2
        }
    }
} // Mocking only necessary

describe('OSliderThumb', () => {
    beforeEach(() => {
        wrapper = shallowMount(OSliderThumb, {
            parentComponent: OSlider
        })
    })

    it('is called', () => {
        expect(wrapper.name()).toBe('OSliderThumb')
        expect(wrapper.isVueInstance()).toBeTruthy()
    })

    it('render correctly', () => {
        expect(wrapper.html()).toMatchSnapshot()
    })

    it('inherits parent properties', () => {
        expect(wrapper.vm.step).toBe(OSlider.data().step)
        expect(wrapper.vm.precision).toBe(OSlider.data().precision)
    })

    it('manage focus', () => {
        wrapper.vm.onFocus()
        expect(wrapper.vm.isFocused).toBeTruthy()

        wrapper.vm.onBlur()
        expect(wrapper.vm.isFocused).toBeFalsy()
    })
})
