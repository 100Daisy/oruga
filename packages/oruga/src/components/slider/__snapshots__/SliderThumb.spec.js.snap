// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OSliderThumb render correctly 1`] = `
<div class="o-slider-thumb-wrapper">
    <o-tooltip-stub active="true" label="0" position="top" triggers="hover" animated="true" animation="fade" autoclose="true" rootclass="o-tooltip" contentclass="o-tooltip-content" triggerclass="o-tooltip-trigger" multilineclass="o-tooltip-multiline" alwaysclass="o-tooltip-always" type="">
        <div tabindex="0" class="o-slider-thumb"></div>
    </o-tooltip-stub>
</div>
`;
